File: 04-CSHARP/08-Funciones/04pasoPorReferencia.cs
// Paso por referencia en C#

// Cambiar valor (Paso por referencia: se pasa la direccion de memoria de la variable original)
void CambiarValor(ref int parametro)
{
    parametro = 5;
}

// Ejemplo de paso por referencia: Se pasa una referencia (Dir. de memoria) de la variable.
int argumento = 10;

Console.WriteLine($"Valor antes de la funcion: {argumento}");
CambiarValor(ref argumento);
Console.WriteLine($"Valor despues de la funcion: {argumento}");

File: 03_CPP/08_funciones/04pasoPorReferencia.cpp
#include <iostream>
using namespace std;

// Cambiar valor (Paso por referencia: se pasa la direccion de memoria de la variable original)
void cambiarValor(int *parametro)
{
    *parametro = 20;
}

int main()
{

    // Ejemplo de paso por referencia: Se pasa una referencia (Dir. de memoria) de la variable.
    int argumento = 10;

    cout << "Valor antes de funcion " << argumento << endl;
    cambiarValor(&argumento);
    cout << "Valor despues de funcion " << argumento << endl;

    return 0;
}